### PR TITLE
Implement setting of user axis range from TPad.

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

File: graf2d/gpadv7/v7/inc/ROOT/TPadUserAxis.hxx [hunk 1]
```diff
@@ -75,6 +49,7 @@ private:
    /// End of the axis range (but see fBoundsKind)
    double fEnd = 1.;
 
+protected:
```
[the remark this answers is not in the document]
Comment (Axel-Naumann) [reply graf2d/gpadv7/v7/inc/ROOT/TPadUserAxis.hxx]: Thanks :-)

File: graf2d/gpadv7/v7/inc/ROOT/TPad.hxx [hunk 2]
```diff
@@ -131,10 +132,27 @@ public:
...
+   TPadUserAxisBase* GetExistingAxis(size_t dimension) const;
+
+   void SetAxisBounds(int dimension, double begin, double end);
+   void SetAxisBound(int dimension, TPadUserAxisBase::EAxisBoundsKind boundsKind, double bound);
+   void SetAxisAutoBounds(int dimension);
+
+   void SetAllAxisBounds(const std::vector<std::array<double, 2>> &vecBeginAndEnd);
+
+   /// Simple struct representing an axis bound.
+   struct BoundKindAndValue {
```
Comment (pcanal) [inline graf2d/gpadv7/v7/inc/ROOT/TPad.hxx]: Is this part of the public interface?

Comment (Axel-Naumann) [reply graf2d/gpadv7/v7/inc/ROOT/TPad.hxx]: Yes. It allows you to specify `SetAllAxisBound({{10., kAxisBoundsEnd}, {0., kAxisBoundsBegin}})`

**Comment by Axel-Naumann** [comment]
@phsft-bot build

File: graf2d/gpadv7/v7/inc/ROOT/TPad.hxx (hunk 2 above)
Comment (pcanal) [reply graf2d/gpadv7/v7/inc/ROOT/TPad.hxx]: Is the default constructor setting the double value to a reasonable value for this purpose?

Also it looks like the code you shown would not compile (is the order of the member correct)?

**Pull request merged**